src/plugin.js

```diff
@@ -762,14 +901,356 @@ void main() {
+            break;
+
+          case 'forward10':
+            this.seekForward10_();
+            break;
+
+          case 'controller':
+            // TODO: drag move controller bar?
+            break;
+
+          case 'exit':
+            this.vrButton.click();
+            if (this.currentSession) {
+              this.currentSession.end();
+              this.player_.pause();
+            }
+            break;
+          }
+          controller.userData.selectPressed = false;
+        }
+        controller.children[0].scale.z = rayTargets[0].distance;
+      } else {
+        this.highlight.visible = false;
+      }
+    }
+  }
+
+  initImmersiveVR() {
+    this.renderer.xr.enabled = true;
+    this.renderer.xr.setReferenceSpaceType('local');
+    this.renderer.setPixelRatio(window.devicePixelRatio);
+    this.renderer.setAnimationLoop(this.render.bind(this));
+
+    this.raycaster = new THREE.Raycaster();
+    this.workingMatrix = new THREE.Matrix4();
+    this.workingVector = new THREE.Vector3();
+
+    this.initShuttleControls();
+
+    const self = this;
+
+    this.controllers = this.buildControllers();
+
+    function onSelectStart() {
+      this.children[0].scale.z = 10;
+      this.userData.selectPressed = true;
+      self.controls.visible = true;
+    }
+
+    function onSelectEnd() {
+      this.children[0].scale.z = 0;
+      self.highlight.visible = false;
+      this.userData.selectPressed = false;
+      self.controls.visible = false;
+    }
+
+    this.controllers.forEach((controller) => {
+      controller.addEventListener('selectstart', onSelectStart);
+      controller.addEventListener('selectend', onSelectEnd);
+      controller.addEventListener('squeezestart', onSelectStart);
+      controller.addEventListener('squeezeend', onSelectEnd);
+      controller.addEventListener('disconnected', onControllerDisconnected);
+    });
+
+    function onControllerDisconnected() {
+      const index = this.userData.index;
+
+      if (self.controllers) {
+        const obj = (index === 0) ? self.controllers[0] : self.controllers[1];
+
+        if (obj) {
+          if (obj.controller) {
+            const controller = obj.controller;
+
+            while (controller.children.length > 0) {
+              controller.remove(controller.children[0]);
+            }
+            self.scene.remove(controller);
+          }
+          if (obj.grip) {
+            self.scene.remove(obj.grip);
+          }
+        }
+      }
+    }
+  }
+
+  render() {
+    if (this.controllers) {
+      const self = this;
+
+      this.controllers.forEach((controller) => {
+        self.renderController(controller);
+      });
+    }
+
+    this.renderer.render(this.scene, this.camera);
+  }
+
+  addCardboardButton_() {
+    // Not using the cardboard button in favour of three.js's vrbutton.
+    // if (!this.player_.controlBar.getChild('CardboardButton')) {
+    //   this.player_.controlBar.addChild('CardboardButton', {});
+    // }
```

Comment:
If the cardboard button is being removed in favor of the three.js VRButton, shouldn't all references to `addCardboardButton_()`, the whole `cardboard-button.js` file, and the `forceCardboard` option be removed? Why go halfway?